### ISSUE TITLE
tf.nn.conv2d_transpose with the same inputs produces different outputs on different calls on GPU

### ISSUE THREAD

**Issue by aiueogawa** [tensorflow/tensorflow#23318]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 16.04.5 LTS (Xenial Xerus)
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: None
- TensorFlow installed from (source or binary): source
- TensorFlow version (use command below): v1.10.1-0-g4dcfddc
- Python version: Python 3.6.5
- Bazel version (if compiling from source): 0.16.1
- GCC/Compiler version (if compiling from source): 5.4.0-6ubuntu1~16.04.10
- CUDA/cuDNN version: V9.0.176/7.1.4.18
- GPU model and memory: GeForce GTX 1080 Ti with 12GB memory

**Describe the current behavior**
`tf.nn.conv2d_transpose` with the same inputs produces different outputs on different calls on GPU.

**Describe the expected behavior**
`tf.nn.conv2d_transpose` with the same inputs is expected to produce the same outputs on different calls on GPU.

**Code to reproduce the issue**
The code below

```
import tensorflow as tf
tf.enable_eager_execution()


x = tf.random_uniform([1, 1024, 32, 32])
filter = tf.random_uniform([4, 4, 96, 1024])

y1 = tf.nn.conv2d_transpose(x, filter, [1, 96, 64, 64], strides=[1, 1, 2, 2], padding='SAME', data_format='NCHW')
y2 = tf.nn.conv2d_transpose(x, filter, [1, 96, 64, 64], strides=[1, 1, 2, 2], padding='SAME', data_format='NCHW')

# show difference between y1 and y2
print(tf.reduce_mean(tf.abs(y1 - y2)))
```

produces a non-zero result like

```
tf.Tensor(8.213489e-06, shape=(), dtype=float32)
```

**Other info / logs**
`tf.nn.conv2d_transpose` with the same inputs produces the same outputs on different calls on **CPU**.

**Comment by ymodak** [comment]
@aselle Assigning to you since you were working on a [similar issue](https://github.com/tensorflow/tensorflow/issues/11882) earlier. I have tested the above snippet in TF 1.12 and it still persists. Thanks!

**Comment by aselle** [comment]
This is not a good match for me. Probably some GPU CudNN expert. Maybe reedwm?

**Comment by reedwm** [comment]
This is likely due to the fact `conv2d_transpose` is non-deterministic on the GPU. @ekelsen, can you confirm?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ekelsen: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by ekelsen** [comment]
Yes, I believe it can be non-deterministic.  But the auto-tuner is also non-deterministic.  So basically all bets are always off right now.

**Comment by jvishnuvardhan** [comment]
@aiueogawa I think it was resolved. As mentioned above, non-deterministic nature of some ops results in minutely different results for the same input on GPU. Please close the issue if it was resolved. Thanks!

**Comment by jvishnuvardhan** [comment]
I think it was resolved. I am closing the issue. Please open a new ticket if you see similar issue again. Thanks!
